internal/controller/environments/environments_test.go

```diff
@@ -40,14 +45,121 @@ func TestNewEnvironmentReconciler(t *testing.T) {
 	require.NotNil(t, e.getLatestCommitIDFn)
 }
 
+func TestIndexEnvsByApp(t *testing.T) {
+	testCases := []struct {
+		name        string
+		environment *api.Environment
+		assertions  func([]string)
+	}{
+		{
+			name: "environment has no health checks",
+			environment: &api.Environment{
+				Spec: &api.EnvironmentSpec{},
+			},
+			assertions: func(res []string) {
+				require.Nil(t, res)
+			},
+		},
+		{
+			name: "environment has health checks",
+			environment: &api.Environment{
+				Spec: &api.EnvironmentSpec{
+					HealthChecks: &api.HealthChecks{
+						ArgoCDAppChecks: []api.ArgoCDAppCheck{
+							{
+								AppNamespace: "fake-namespace",
+								AppName:      "fake-app",
+							},
+							{
+								AppNamespace: "another-fake-namespace",
+								AppName:      "another-fake-app",
+							},
+						},
+					},
+				},
+			},
+			assertions: func(res []string) {
+				require.Equal(
+					t,
+					[]string{
+						"fake-namespace:fake-app",
+						"another-fake-namespace:another-fake-app",
+					},
+					res,
+				)
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			indexEnvsByApp(testCase.environment)
+		})
+	}
+}
```

Comment:
Admittedly, this test isn't directly related to this PR, but see [this comment](https://github.com/akuity/kargo/pull/341/files#r1179671413) about a small bit of refactoring. It made this function testable, where it was less so before.